### PR TITLE
MM-47290 : Migrate "/channel_sidebar" e2e tests to TypeScript

### PR DESCRIPTION
**Pull request by cinlloc**
#### Summary
Migrated channel_sidebar e2e tests to TypeScript

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/21297

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by furqanmlk — COMMENTED** [review]
thanks @tbalthazar 
looks good to me just small suggestion

File: e2e/cypress/tests/integration/channel_sidebar/category_muting_spec.ts [hunk 1]
```diff
@@ -10,6 +10,8 @@
 // Stage: @prod
 // Group: @channel_sidebar
 
+import {Channel} from '@mattermost/types/lib/channels';
```
[the remark this answers is not in the document]
Comment (cinlloc) [reply e2e/cypress/tests/integration/channel_sidebar/category_muting_spec.ts]: Sure. Fixed in [a35fd15](https://github.com/mattermost/mattermost-webapp/pull/11392/commits/a35fd15bd51446f88be43c61db6f5e5173717cc8)

**Review by furqanmlk — APPROVED** [review]
Thanks @cinlloc

**Review by saturninoabril — APPROVED** [review]
Thanks @cinlloc!

**Pull request merged**